pkg/controller/multiclusterstatusaggregation/multiclusterStatusAggregation_controller.go

```diff
@@ -570,47 +571,34 @@ func (r *ReconcilePullModelAggregation) cleanupOrphanReports() error {
 	}
 
 	for m := range missingAppset {
-		// Look for ManifestWorks associated with this AppSet and then removed them
 		appSet := &missingAppset[m]
+		errOccured := false
 
-		appSetHash, err := propagation.GenerateManifestWorkAppSetHashLabelValue(appSet.Namespace, appSet.Name)
-		if err != nil {
-			klog.Errorf("error generating appSet hash: %v", err)
-			continue
-		}
-
-		appSetRequirement, err := labels.NewRequirement(propagation.LabelKeyAppSetHash, selection.Equals, []string{appSetHash})
-		if err != nil {
-			klog.Errorf("bad requirement: %v", err)
-			continue
-		}
+		if appSet.Statuses.ClusterConditions != nil && len(appSet.Statuses.ClusterConditions) > 0 {
+			for _, clusterCondition := range appSet.Statuses.ClusterConditions {
+				if clusterCondition.App != "" {
+					nsn := strings.Split(clusterCondition.App, "/")
+					workNamespace := nsn[2]
+					workName := nsn[3]
```

Comment:
It would be better to validate if nsn[2] and nsn[3] exists before assignment. This is to avoid go panic in some corner cases

Reply:
Done as suggested in the new commit: https://github.com/open-cluster-management-io/multicloud-integrations/pull/77/commits/1ed7f45fd69b1e0e725eb35c9759337d22d8e163